### PR TITLE
Store web hook & query result in args object

### PR DESCRIPTION
**Pull request by sharadregoti**

### CODE REVIEW

File: gateway/modules/auth/match.go [hunk 1]
```diff
@@ -80,7 +81,18 @@ func (m *Module) matchFunc(ctx context.Context, rule *config.Rule, MakeHTTPReque
 	}
 
 	var result interface{}
-	return formatError(ctx, rule, MakeHTTPRequest(ctx, "POST", rule.URL, token, scToken, obj, &result))
+	if err := formatError(ctx, rule, MakeHTTPRequest(ctx, http.MethodPost, rule.URL, token, scToken, obj, &result)); err != nil {
```
Comment (YourTechBud) [inline gateway/modules/auth/match.go:84]: Call `formatError` only if `MakeHTTPRequest` returns an error

File: gateway/modules/auth/match.go [hunk 2]
```diff
@@ -80,7 +81,18 @@ func (m *Module) matchFunc(ctx context.Context, rule *config.Rule, MakeHTTPReque
...
+		return err
+	}
+
+	if rule.Store != "" {
```
Comment (YourTechBud) [inline gateway/modules/auth/match.go:88]: If store value is not provided, store the result in `args.result` by default.

File: gateway/modules/auth/match.go [hunk 3]
```diff
@@ -96,9 +108,14 @@ func (m *Module) matchQuery(ctx context.Context, project string, rule *config.Ru
...
-		return nil, formatError(ctx, rule, err)
+	if rule.Store != "" {
+		if err := utils.StoreValue(ctx, rule.Store, data, args); err != nil {
+			return nil, formatError(ctx, rule, err)
+		}
+	} else {
+		if err := utils.StoreValue(ctx, "args.result", data, args); err != nil {
+			return nil, formatError(ctx, rule, err)
+		}
 	}
```
Comment (YourTechBud) [inline gateway/modules/auth/match.go:119]: You can improve on the logic. 
Set the value of  `rule.Store` to `args.result` when empty.

**Review by YourTechBud — CHANGES_REQUESTED** [review]
Let me know if you've tested this

**Comment by sharadregoti** [comment]
> Let me know if you've tested this

Yes i did

**Review by YourTechBud — APPROVED** [review]

**Pull request merged**